android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -1551,20 +1551,23 @@ public void onNavigationStarted()
     ThemeSwitcher.INSTANCE.restart(isMapRendererActive());
     mNavigationController.start(this);
     initNavigationButtons(MapButtonsController.LayoutMode.navigation);
+    refreshLightStatusBar();
   }
 
   @Override
   public void onPlanningCancelled()
   {
     closeFloatingToolbarsAndPanels(true);
     initNavigationButtons(MapButtonsController.LayoutMode.regular);
+    refreshLightStatusBar();
```

Comment:
nit: Is it the best way to refresh? Can copy-paste be avoided in a more beautiful way?

Btw, why it should be refreshed every time here, in every callback?

Reply:
The best way would be to create different fragments for planning/navigation modes. We could then setup the status bar in those fragments instead of relying on callbacks.

As it is not fragments, we need to make sure the status bar color is correctly updated when planning/navigation starts/ends (black text for navigation and white for planning).